src/MusicalScore/Graphical/SkyBottomLineCalculator.ts

```diff
@@ -152,6 +164,43 @@ export class SkyBottomLineCalculator {
         this.mBottomLine = this.mBottomLine.map(v => (v - Math.min(...this.mBottomLine)) / unitInPixels + this.StaffLineParent.BottomLineOffset);
     }
 
+    /**
+     * go backwards through the skyline array and find a number so that
+     * we can properly calculate the average
+     * @param start
+     * @param backend
+     * @param color
+     */
+    private findPreviousValidNumber(start: number, tSkyLine: number[]): number {
+        for (let idx: number = start; idx >= 0; idx--) {
+            if (!isNaN(tSkyLine[idx]) && tSkyLine[idx] !== undefined) {
+                return tSkyLine[idx];
+            } else {
```

Comment:
remove the else branch. this would always return 0 if the first previous value is undefined. but we want to keep looking until we find a better previous value.
we now return 0 after the for loop, that's why we don't need the else anymore.

Reply:
This is already fixed in findNextValidNumber, it looks like.